task_2/task_2.txt

```diff
@@ -0,0 +1,23 @@
+There are several possible ways to provide the secrets to running pods securely:
+
+1. AWS KMS envelope encryption https://aws.amazon.com/blogs/containers/using-eks-encryption-provider-support-for-defense-in-depth/;
++ completely transparent to the application;
++ does not break common flow, secrets are referenced as usual;
++ no additional configuration is needed;
```

Comment:
How would your "secrets management pipeline / workflow" look like in that case? Let's assume I'm a developer working in that team, and I need to provide a database password for my app running on an EKS cluster, one in staging and one in production. How KMS envelope encryption will help me with that workflow?

Reply:
Envelope encryption is transparent to the app and users so the workflow would be the following:
- generate secret with kubectl;
- reference this secret in the pod definition;
- EKS performs encryption under the hood;
- when pod is created, referenced secrets are mounted/injected as env variables as usually.
does it make sense?

Reply:
Definitely makes sense, @eugenetaranov. My only concern is that with that approach we don't have any secrets management "features". For example - no audit log on who has changed the secret and when. No backup in case the cluster goes down.

So it's definitely an option, and we can secure our secrets by using the envelope encryption, but we won't get any "secrets management" stuff since Kubernetes secrets are lacking that part.